syntax = "proto3";
option cc_generic_services =true;
package rpcService;

message AppendEntriesRequest {
    int64 term  = 1;
    string leaderId  = 2;
    int64 prevLogIndex = 3;
    int64 prevLogTerm =4;


    message LogEntry{
        int64 index = 1;
        int64 term = 2;
        string commandName =3 ;
//        bytes command = 4;
    }
    repeated LogEntry LogEntries = 5;
    int64 leaderCommit =6 ;
}

message AppendEntriesResponse {
    int64 Term  = 1;
    int64 Index = 2;
    int64 CommitIndex =3;
    bool Success  = 4;
}

message RequestVoteRequest {
     uint64 Term=1;
     uint64 LastLogIndex=2;
     uint64 LastLogTerm=3;
     string CandidateName=4;
}


message RequestVoteResponse {
     uint64 Term=1;
     bool VoteGranted=2;
}


service RaftService {
    rpc AppendEntries(AppendEntriesRequest) returns(AppendEntriesResponse);
    rpc Vote(RequestVoteRequest) returns(RequestVoteResponse);
}